src/main/java/org/reaktivity/command/log/internal/LoggableStream.java

```diff
@@ -149,52 +211,23 @@ private boolean handleFrame(
             return false;
         }
 
-        switch (msgTypeId)
+        MessageConsumer handler = frameHandlers.get(msgTypeId);
+        if (handler != null)
         {
-        case BeginFW.TYPE_ID:
-            final BeginFW begin = beginRO.wrap(buffer, index, index + length);
-            onBegin(begin);
-            break;
-        case DataFW.TYPE_ID:
-            final DataFW data = dataRO.wrap(buffer, index, index + length);
-            onData(data);
-            break;
-        case EndFW.TYPE_ID:
-            final EndFW end = endRO.wrap(buffer, index, index + length);
-            onEnd(end);
-            break;
-        case AbortFW.TYPE_ID:
-            final AbortFW abort = abortRO.wrap(buffer, index, index + length);
-            onAbort(abort);
-            break;
-        case ResetFW.TYPE_ID:
-            final ResetFW reset = resetRO.wrap(buffer, index, index + length);
-            onReset(reset);
-            break;
-        case WindowFW.TYPE_ID:
-            final WindowFW window = windowRO.wrap(buffer, index, index + length);
-            onWindow(window);
-            break;
-        case SignalFW.TYPE_ID:
-            final SignalFW signal = signalRO.wrap(buffer, index, index + length);
-            onSignal(signal);
-            break;
-        case ChallengeFW.TYPE_ID:
-            final ChallengeFW challenge = challengeRO.wrap(buffer, index, index + length);
-            onChallenge(challenge);
-            break;
-        case FlushFW.TYPE_ID:
-            final FlushFW flush = flushRO.wrap(buffer, index, index + length);
-            onFlush(flush);
-            break;
+            handler.accept(msgTypeId, buffer, index, length);
         }
 
         return true;
     }
 
     private void onBegin(
-        final BeginFW begin)
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
     {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+
```

Comment:
Prefer to keep these method signatures strongly typed, so let's register the wrap logic in the frame handler map instead.

```
if (frameTypes.test("BEGIN"))
{
    frameHandlers.put(BeginFW.TYPE_ID, (t, b, i, l) -> onBegin(beginRO.wrap(t, b, i, i + l));
}
```
same for other frame type method signatures.